theano/tensor/basic.py

```diff
@@ -6093,3 +6093,12 @@ def c_code_cache_version(self):
 
     def do_constant_folding(self, node):
         return False
+
+    def connection_pattern(self, node):
+        return [[False] for i in node.inputs]
+
+    def grad(self, inputs, grads):
+        return [DisconnectedType()() for i in inputs]
+
+    def R_op(self, inputs, eval_points):
+        return [zeros(inputs, self.dtype)]
```

Comment:
This is probably because `IncSubtensor.R_op` returns `None` if _either_ differentiable input is `None`, whereas it should only do so if _both_ are.
Since there may be other Ops doing the same error, I think we should be on the side of correct results, rather than maximum speed. So I agree with your current implementation.